plugins/generic/dublinCoreMeta/DublinCoreMetaPlugin.php

```diff
@@ -188,20 +193,21 @@ public function monographView($hookName, $args)
      */
     public function monographFileView($hookName, $args)
     {
-        $request = Application::get()->getRequest();
         $monograph = $args[1];
         $publicationFormat = $args[2];
         $submissionFile = $args[3];
-        $requestArgs = $request->getRequestedArgs();
-        $press = $request->getContext();
+
+        $publication = $monograph->getCurrentPublication();
 
         // Only add Google Scholar metadata tags to the canonical URL for the latest version
         // See discussion: https://github.com/pkp/pkp-lib/issues/4870
-        if (count($requestArgs) > 1 && $requestArgs[1] === 'version') {
+        if ($publicationFormat->getData('publicationId') != $publication->getId()) {
```

Comment:
I realized that there will be no 'version' in URL when a file is viewed, so I am checking this way if it is a file i.e. publication format of a current version/publication.